gcp/appengine/blog/content/posts/introducing-broad-c-c++-support/index.md

```diff
@@ -0,0 +1,89 @@
+<!--more-->
+
+Because the C/C++ ecosystem doesn’t have a centralized package register, source code identifiers (e.g. git hashes) are the best way to identify C/C++ libraries. Typically, vulnerabilities are associated with versions, not Git hashes, making C/C++ vulnerability matching difficult. The new commit level vulnerability information will allow users to confidently match their dependencies to known vulnerabilities within the OSV database. 
+
+Vulnerable commit ranges provide a granular approach that more closely follows the development process, matching vulnerabilities to a range of upstream commits instead of matching vulnerabilities to a package version. This precise information allows for accurate identification of Git branches cut from known vulnerable commits and leads to more accurate vulnerability information.
+
+## How are vulnerable commit ranges included in the OSV database?
+
+We’ve enriched over 30,000 advisories based on the [NVD CVE database](https://nvd.nist.gov/vuln/search). We’ve added vulnerable commit ranges to these advisories by using algorithmic analysis to match Git repository tag information to patch information supplied by the NVD CVE entry. 
+
+As part of our work in this area, we've established working relationships with the NVD via data quality fixes and longer term systematic improvements.
+
+![Image shows the vulnerable commit ranges for CVE-2023-26130. More information is available in the "Submoduled C/C++ dependencies" section.](commit-range.png "Vulnerable commit ranges for CVE-2023-26130")
+Sample OSV advisory with commit range information.
+
+## Finding C/C++ vulnerabilities with vulnerable commit ranges
+
+Matching C/C++ dependencies to known vulnerabilities has been one of the final pieces in the puzzle of a truly comprehensive open source vulnerability database. Most projects with C/C++ dependencies include a copy of those dependencies bundled with the project, either by using submodules or by vendoring dependencies. Whether dependencies are submoduled or vendored, Vulnerable commit ranges allow OSV users to match their dependencies to known vulnerabilities. 
+
+### Submoduled C/C++ dependencies
+
+Submoduled dependencies retain their Git histories. With a few simple Git commands, it’s possible to know the current Git commit hash for every dependency. Once the relevant Git hash is known, we can use the [OSV API](https://google.github.io/osv.dev/post-v1-query/) to find any corresponding vulnerabilities. 
+
+For example, let’s take a look at the dependencies for the [pd-server](https://github.com/charlesneimog/pd-server) project and see if we can find any vulnerabilities. Pd-server is a PureData interface to cpp-httplib and includes cpp-httplib as a submoduled dependency. 
```

Comment:
Let's perhaps use something like https://github.com/yuzu-emu/yuzu as the example? It's a bit higher profile. @andrewpollock thoughts?

Reply:
@oliverchang I think I would prefer to feature yuzu, but I wonder if it is an issue that the vuln is in the most recent version. For pd-server, the issue is resolved. 

@andrewpollock

Reply:
Yeah I can see the attractiveness of using yuzu, in that that repo is more high-profile. 

The intention of the example was to demonstrate finding vulnerabilities in submodules, and the existing example is a random one I found that was using `cpp-httplib` as a submodule. That's meant to the be interesting part.

We're back at the "we're going to 0-day a repo" with the blog post, I guess let's see if we can get this one fixed as well? The fact that they've got 1.4k open issues doesn't fill me with hope...

Reply:
FYI I've filed https://github.com/yuzu-emu/yuzu/issues/11919

Reply:
Thanks! I think it does make the messaging stronger if we show this is useful for high profile projects.

Re 0-day a repo, I do think running a public scanner against known public vulnerabilities (CVEs) is quite far away from dropping an exploit. Looks like they're already on it though (https://github.com/yuzu-emu/yuzu/pull/11728) :) so win-win!

Reply:
The issue should be fixed in the most recent build of yuzu mainline as of writing (1605), as the dependency updates were tagged for inclusion for today's build.

Reply:
So it should now be perfectly fine to switch the example over to use https://github.com/yuzu-emu/yuzu instead...